modules/tcfControl.js

```diff
@@ -367,8 +369,25 @@ export function setEnforcementConfig(config) {
   }
 }
 
+export function checkIfCredentialsAllowed(next, options = {}, moduleType, moduleName) {
+  if (!options.withCredentials || (moduleType && moduleName)) {
+    next(options);
+    return;
+  }
+  const consentData = gdprDataHandler.getConsentData();
+  const rule = ACTIVE_RULES.purpose[1];
+  const ruleOptions = CONFIGURABLE_RULES[rule.purpose];
+  const {purpose} = getConsent(consentData, ruleOptions.type, ruleOptions.id, null);
```

Comment:
IMO you do need to split the validateRules logic. For example if I set `enforcePurpose: false` on the storage rule I would expect this check to be turned off.